inc/commoninjectionlib.class.php

```diff
@@ -1513,13 +1513,14 @@ private function effectiveAddOrUpdate($injectionClass, $item, $values, $add = tr
 
       foreach ($values as $key => $value) {
          $option = self::findSearchOption($options, $key);
-         if ($key === 'id' || $option !== false && (!isset($option['checktype']) || $option['checktype'] != self::FIELD_VIRTUAL)) {
-            //If field is a dropdown and value is '', then replace it by 0
-            if ($option !== false && self::isFieldADropdown($option['displaytype']) && $value == self::EMPTY_VALUE) {
-               $toinject[$key] = self::DROPDOWN_EMPTY_VALUE;
-            } else {
-               $toinject[$key] = $value;
-            }
+         if ($option === false) {
+             $toinject[$key] = $value;
+         } else if (!isset($option['checktype']) || $option['checktype'] != self::FIELD_VIRTUAL) {
+             if (self::isFieldADropdown($option['displaytype']) && $value == self::EMPTY_VALUE) {
+                 $toinject[$key] = self::DROPDOWN_EMPTY_VALUE;
+             } else {
+                 $toinject[$key] = $value;
+             }
          }
```

Comment:
Would be probably easier to read.
```suggestion
         if ($option !== false && isset($option['checktype']) && $option['checktype'] == self::FIELD_VIRTUAL) {
             break;
         }

         if ($option !== false && self::isFieldADropdown($option['displaytype']) && $value == self::EMPTY_VALUE) {
             //If field is a dropdown and value is '', then replace it by 0
             $toinject[$key] = self::DROPDOWN_EMPTY_VALUE;
         } else {
             $toinject[$key] = $value;
         }
```